packages/dds/map/src/test/mocha/map.spec.ts

```diff
@@ -773,6 +777,133 @@ describe("Map", () => {
 		});
 	});
 
+	describe("Attribution", () => {
+		describe("Local State", () => {
+			let map: SharedMap;
+
+			beforeEach(async () => {
+				const options: IMapOptions = { attribution: { track: true } };
+				map = createLocalMap("testMap", options);
+			});
+
+			it("Should not generate attribution in local state", () => {
+				map.set("first", "second");
+				map.set("third", "fourth");
+				map.set("fifth", "sixth");
+
+				const summaryContent = (map.getAttachSummary().summary.tree.header as ISummaryBlob)
+					.content;
+
+				assert.equal(
+					map.getAllAttribution()?.size,
+					0,
+					"do not have correct number of attribution",
+				);
+				assert.equal(
+					summaryContent,
+					`{"blobs":[],"content":{"first":{"type":"Plain","value":"second"},"third":{"type":"Plain","value":"fourth"},"fifth":{"type":"Plain","value":"sixth"}}}`,
+				);
+			});
+		});
+
+		describe("Connected State", () => {
+			let containerRuntimeFactory: MockContainerRuntimeFactory;
+			let map1: SharedMap;
+			let map2: SharedMap;
+
+			beforeEach(async () => {
+				const options: IMapOptions = { attribution: { track: true } };
+				containerRuntimeFactory = new MockContainerRuntimeFactory();
+				// Create the first map
+				map1 = createConnectedMap("map1", containerRuntimeFactory, options);
+				// Create and connect a second map
+				map2 = createConnectedMap("map2", containerRuntimeFactory, options);
+			});
+
+			it("Can generate proper attribution in connected state", () => {
+				map1.set("first", "value1");
+				map1.set("second", "value2");
+				map2.set("first", "value3");
+
+				containerRuntimeFactory.processSomeMessages(1);
+
+				assert.notEqual(
```

Comment:
assert on what the attribution *should* be, not what it shouldn't be. you should be able to spec out which seq it should be attributed to here.

Reply:
same for the assertion below.